libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -548,6 +548,19 @@ void AsmAnalyzer::warnOnInstructions(solidity::Instruction _instr, SourceLocatio
 			"the Metropolis hard fork. Before that it acts as an invalid instruction."
 		);
 
+	static set<solidity::Instruction> experimentalInstructions{
+		solidity::Instruction::SHL,
+		solidity::Instruction::SHR,
+		solidity::Instruction::SAR
+	};
+	if (experimentalInstructions.count(_instr))
+		m_errorReporter.warning(
+			_location,
+			"The \"" +
+			boost::to_lower_copy(instructionInfo(_instr).name)
+			+ "\" instruction is experimental and not available in regular clients. "
```

Comment:
Blockchains instead of clients? Ethereum implementations?

Reply:
Well since it was designated for constantinople we could just update the text to reflect that.